lib/graphql/schema.rb

```diff
@@ -1670,6 +1670,36 @@ def did_you_mean(new_dym = NOT_CONFIGURED)
         end
       end
 
+
+      # If you need to support previous, non-spec behavior which allowed selecting union fields
+      # but *not* selecting any fields on that union, set this to `true` to continue allowing that behavior.
+      #
+      # If this is `true`, then {.legacy_invalid_empty_selections_on_union} will be called with {Query} objects
+      # with that kind of selections. You must implement that method
+      # @param new_value [Boolean]
+      # @return [true, false, nil]
+      def allow_legacy_invalid_empty_selections_on_union(new_value = NOT_CONFIGURED)
```

Comment:
These are probably worth noting in the guide docs given that you have to invoke the setting on your schema or experience an out-of-the-box warning. I suppose the future state defaults this to `true`, so maybe the oddity is sufficiently short-lived.

Reply:
Yes, it's a bummer to have a warning as the default behavior. But my guess is that this edge case is rarely encountered, since other GraphQL implementations (rightly) reject this -- like GraphiQL, for example.

Reply:
Last I looked, we had about 30K / 5K RPM of these two cases (conflicting types / missing union selections), so plenty of clients in the world are running without client-side validation unfortunately.

Reply:
Good question. Those observations were from over a year ago; I want to say there was ~6M RPM on that API back in those days? So yeah, not a significant total share, but still a lot of traffic. I'm setting up new metrics for this update so we can get a fresh sense of quantity and, more importantly, how many consumers they're coming from.

Reply:
The other interesting this would be how many _unique_ query strings. It could be so few query strings that you could basically hardcode a replacement query string for known "offenders" and then call it day.